itest/lnd_coop_close_rbf_test.go

```diff
@@ -0,0 +1,135 @@
+	// initiated RBF attempts will always be successful.
+	ht.SetFeeEstimate(250)
+	ht.SetFeeEstimateWithConf(250, 6)
+
+	// To kick things off, we'll create two new nodes, then fund them with
+	// fund enough coins to make a 50/50 channel.
+	cfgs := [][]string{rbfCoopFlags, rbfCoopFlags}
+	params := lntest.OpenChannelParams{
+		Amt:     btcutil.Amount(1000000),
+		PushAmt: btcutil.Amount(1000000 / 2),
+	}
+	chanPoints, nodes := ht.CreateSimpleNetwork(cfgs, params)
+	alice, bob := nodes[0], nodes[1]
+	chanPoint := chanPoints[0]
+
+	// Now that both sides are active with a funded channel, we can kick
+	// off the test.
+	//
+	// To start, we'll have Alice try to close the channel, with a fee rate
+	// of 5 sat/byte.
+	aliceFeeRate := chainfee.SatPerVByte(5)
+	aliceCloseStream, aliceCloseUpdate := ht.CloseChannelAssertPending(
+		alice, chanPoint, false,
+		lntest.WithCoopCloseFeeRate(aliceFeeRate),
+		lntest.WithLocalTxNotify(),
+	)
+
+	// Confirm that this new update was at 5 sat/vb.
+	alicePendingUpdate := aliceCloseUpdate.GetClosePending()
+	require.NotNil(ht, aliceCloseUpdate)
+	require.Equal(
+		ht, int64(aliceFeeRate), alicePendingUpdate.FeePerVbyte,
+	)
+	require.True(ht, alicePendingUpdate.LocalCloseTx)
+
+	// Now, we'll have Bob attempt to RBF the close transaction with a
+	// higher fee rate, double that of Alice's.
+	bobFeeRate := aliceFeeRate * 2
+	bobCloseStream, bobCloseUpdate := ht.CloseChannelAssertPending(
+		bob, chanPoint, false, lntest.WithCoopCloseFeeRate(bobFeeRate),
+		lntest.WithLocalTxNotify(),
+	)
+
+	// Confirm that this new update was at 10 sat/vb.
+	bobPendingUpdate := bobCloseUpdate.GetClosePending()
+	require.NotNil(ht, bobCloseUpdate)
+	require.Equal(ht, bobPendingUpdate.FeePerVbyte, int64(bobFeeRate))
+	require.True(ht, bobPendingUpdate.LocalCloseTx)
+
+	var err error
+
+	// Alice should've also received a similar update that Bob has
+	// increased the closing fee rate to 10 sat/vb with his settled funds.
+	aliceCloseUpdate, err = ht.ReceiveCloseChannelUpdate(aliceCloseStream)
+	require.NoError(ht, err)
+	alicePendingUpdate = aliceCloseUpdate.GetClosePending()
+	require.NotNil(ht, aliceCloseUpdate)
+	require.Equal(ht, alicePendingUpdate.FeePerVbyte, int64(bobFeeRate))
+	require.False(ht, alicePendingUpdate.LocalCloseTx)
+
+	// We'll now attempt to make a fee update that increases Alice's fee
+	// rate by 6 sat/vb, which should be rejected as it is too small of an
+	// increase for the RBF rules. The RPC API however will return the new
+	// fee. We'll skip the mempool check here as it won't make it in.
+	aliceRejectedFeeRate := aliceFeeRate + 1
+	_, aliceCloseUpdate = ht.CloseChannelAssertPending(
+		alice, chanPoint, false,
+		lntest.WithCoopCloseFeeRate(aliceRejectedFeeRate),
+		lntest.WithLocalTxNotify(), lntest.WithSkipMempoolCheck(),
+	)
+	alicePendingUpdate = aliceCloseUpdate.GetClosePending()
+	require.NotNil(ht, aliceCloseUpdate)
+	require.Equal(
+		ht, alicePendingUpdate.FeePerVbyte,
```

Comment:
This got me to think maybe there should be a field to tell the user the RBF is failed? sth like `Succeeded`?

Reply:
Issue with that is that we can't always detect if it failed or not. For Neutrino, we have no idea. One option we have is to always try to do a fee bump if the user wants one. The side effect of that is that we may end up paying more than the user expect, eg: they try to do 5 sat/vb, but 8 sat/vb is actually what'll trigger a replacement. 

I'm open to modifying the broadcast flow in another PR. It'll need some changes in the `protofsm` executor, and reuslt in some additional states in the state machine (`BroadcastPending` -> `BroadcastFailed` -> `BroadcastSuccess`, etc). I want to restrict this PR to just fundamental blockers to merge, as we still need to do further interop, and get the rc in user's hands so we can get feedback on the RPC/CLI changes.